display_list/display_list.h

```diff
@@ -311,8 +309,6 @@ class DisplayList : public SkRefCnt {
     return modifies_transparent_black_;
   }
 
-  const DisplayListStorage& GetStorage() const { return storage_; }
-
```

Comment:
I think the crux of your argument is around exposing this immutable data ( https://github.com/flutter/engine/pull/52705#discussion_r1597226164 ).  That decision is tied to https://github.com/flutter/engine/pull/52715 since it accesses it.

Everything else in this PR is correct since it statically asserts correct constness, so you can make the revert just the removal of this method.

Reply:
There is nothing wrong with the current implementation. Adding const in a few places is probably fine, but not needed since the storage is not exposed outside of the internal implementation.

Two questions I have is whether or not destructors are required to work on const objects (I supposed that they are or the PR would not have landed) and whether or not the internal const-ness (which isn't strictly necessary) would imply that the data in the buffer never changes (but it actually does - there are mutable fields in the data that provided required functions).

Reply:
But, yes, the rest of it doesn't hurt anything and might be considered cleaner other than the fact that the data stream actually contains at least one (and potentially more as we migrate the text stuff to be Skia/Impeller-agnostic) mutable chunk of data.

Reply:
If we need mutable storage in DisplayList the correct solution would to make `DisplayList::storage_` not const.  Previously it was declared `const` but had the ability to mutate it because there was a const method that returned a non-const pointer.

Reply:
It won't matter as the mutation happens on a const object contained within the DL that is passed to a Dispatch receiver - and it's the receiver that does the mutation.

My point was less "how are we going to do this" than "we do this and making this pointer const won't stop it, so are we sending the right message by making it const", but I think it is a minor point that doesn't matter...

Reply:
The fact that the field is marked "mutable" already acknowledges that it is allowing mutation inside what is otherwise a const set of data, so I think the point is moot.